src/transaction.rs

```diff
@@ -82,38 +82,45 @@ pub struct Transaction {
     /// A digital signature of `instruction`, `last_id` and `fee`, signed by `Pubkey`.
     pub signature: Signature,
 
-    /// The `Pubkey` of the entity that signed the transaction data.
-    pub from: Pubkey,
-
-    /// The action the server should take.
-    pub instruction: Instruction,
+    /// The `Pubkeys` that are executing this transaction userdata.  The meaning of each key is
+    /// contract-specific.
+    /// * keys[0] - Typically this is the `from` public key.  `signature` is verified with keys[0].
```

Comment:
"Typically"?  In which cases is this not true?  That may be helpful to highlight?

Reply:
In the case that it's implement the Budget DSL.  @aeyakovenko, it's strange to mention the `from` parameter here. You'd need historical context to grok it. This is a tough one to describe. Thoughts?